tests/modules/programs/yazi/fish-integration-enabled.nix

```
@@ -1,27 +1,18 @@
 { ... }:
 
-let
-  shellIntegration = ''
-    function yy
-      set tmp (mktemp -t "yazi-cwd.XXXXX")
-      yazi $argv --cwd-file="$tmp"
-      if set cwd (cat -- "$tmp"); and [ -n "$cwd" ]; and [ "$cwd" != "$PWD" ]
-        builtin cd -- "$cwd"
-      end
-      rm -f -- "$tmp"
-    end
-  '';
-in {
+{
   programs.fish.enable = true;
 
   programs.yazi = {
     enable = true;
+    shellWrapperName = "yy";
     enableFishIntegration = true;
   };
 
   test.stubs.yazi = { };
 
   nmt.script = ''
-    assertFileContains home-files/.config/fish/config.fish '${shellIntegration}'
+    assertFileContent home-files/.config/fish/functions/yy.fish \
```

I'd personally prefer if we don't hardcode the function name, here, since the test can break if it's not changed when the `shellWrapperName` does.

```suggestion
    assertFileContent home-files/.config/fish/functions/${config.programs.yazi.shellWrapperName}.fish \
```

---

Good point, I've amended the commit with this change.